monkey/infection_monkey/network/tcp_scanner.py

```diff
@@ -58,31 +59,18 @@ def _check_tcp_ports(ip: str, ports: List[int], timeout: float = DEFAULT_TIMEOUT
                 connected_ports_sockets.append((port, sock))
                 possible_ports.append((port, sock))
                 continue
-            # BUG: I don't think a socket will ever connect successfully if this error is raised.
-            #      From the documentation: "Resource temporarily unavailable... It is a nonfatal
-            #      error, **and the operation should be retried later**." (emphasis mine). If the
-            #      operation is not retried later, I don't see the point in appending this to
-            #      possible_ports.
-            if err == 10035:  # WSAEWOULDBLOCK is valid, see
-                # https://msdn.microsoft.com/en-us/library/windows/desktop/ms740668%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396
-                possible_ports.append((port, sock))
-                continue
             if err == 115:  # EINPROGRESS     115     /* Operation now in progress */
                 possible_ports.append((port, sock))
                 continue
             logger.warning("Failed to connect to port %s, error code is %d", port, err)
 
         if len(possible_ports) != 0:
             timeout = int(round(timeout))  # clamp to integer, to avoid checking input
-            sockets_to_try = possible_ports[:]
-            # BUG: If any sockets were added to connected_ports_sockets on line 94, this would
-            #      remove them.
-            connected_ports_sockets = []
+            sockets_to_try = possible_ports.copy()
             while (timeout >= 0) and sockets_to_try:
                 sock_objects = [s[1] for s in sockets_to_try]
 
-                # BUG: Since timeout is 0, this could block indefinitely
-                _, writeable_sockets, _ = select.select(sock_objects, sock_objects, sock_objects, 0)
+                _, writeable_sockets, _ = select.select([], sock_objects, [], 0)
```

Comment:
Let's create a constant at the top of the file;

```
NEVER_BLOCK = 0
```

and then

```suggestion
                _, writeable_sockets, _ = select.select([], sock_objects, [], NEVER_BLOCK)
```

Reply:
I've resolved this using a timer as mentioned in https://github.com/guardicore/monkey/pull/1705#discussion_r802959847. Please review :)